app/pages/clinicdetails/clinicdetails.js

```diff
@@ -169,14 +173,21 @@ export const ClinicDetails = (props) => {
           // clinic patients have not been migrated, we open the prompt to complete the migration
           openMigrationConfirmationModal();
         } else {
-          if (action !== 'new' || (action === 'new' && previousSubmitting)) {
+          if (action !== 'new') {
             // If there is no reason for the user to be here, we redirect them appropriately
             redirectToWorkspace();
           }
         }
       }
     }
-  }, [clinic, submitting]);
+  }, [
+    action,
+    clinic,
+    submitting,
+    previousSubmitting,
```

Comment:
`previousSubmitting` is no longer used and can be excluded (though should only change with `submitting`, so it's likely harmless). this is the only reference to the variable outside of declaration, so we could drop it from the file entirely I think.

Reply:
Good call.  Will remove